src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private final Random rand = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int number = rand.nextInt(FIGURE_COUNT);
+        String randomColor = colorSupplier.getRandomColor();
+        switch (number) {
+            case 0:
+                double radius = rand.nextDouble(100.00);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example: 

double radius = rand.nextDouble() * 100;
// ... 
return new Circle(randomColor, radius);